### PR TITLE
Clean-up fixes before 2022-Dec release

### PR DESCRIPTION
**Pull request by jfingerh**

### CODE REVIEW

File: pna.p4 [hunk 1]
```diff
@@ -574,16 +574,20 @@ struct pna_main_parser_input_metadata_t {
...
+extern bool is_net_port (in PortId_t p);
 
 struct pna_main_input_metadata_t {
     // common fields initialized for all packets that are input to main
     // parser, regardless of direction.
     bool                     recirculated;
     Timestamp_t              timestamp;
     ParserError_t            parser_error;
-    ClassOfService_t         class_of_service; //TODO: is this to stay?
+    ClassOfService_t         class_of_service;
```
Comment (jfingerh) [inline pna.p4:590]: Change recommended by Thomas Calvert in https://github.com/p4lang/pna/pull/104

**Comment by apinski-cavium** [comment]
LGTM

File: PNA.mdk [hunk 2]
```diff
@@ -580,38 +580,16 @@ extern void send_to_port(PortId_t dest_port);
...
-
-+ current packet direction is `NET_TO_HOST`, and `x` is a network port id.
-
-Calling `send_to_port(x)` modifies hidden state for this packet, so
-that when the packet is finished with the main control and main
-deparser, it will loop back in the host side, and later return to
-be processed by the main control in the HOST_TO_NET direction.  The
-hidden state will remain associated with the packet during that
-loopback, so that even if no further forwarding functions are
-called for the packet, it will be transmitted out of network port
```
Comment (jfingerh) [inline PNA.mdk:600]: This deleted text and the next "hunk" deleted below eliminate some remaining mentions of NET_TO_HOST and HOST_TO_NET, and also simplify the description of `send_to_port` based upon the latest changes made elsewhere in the PNA spec.

File: examples/ipsec-acc.p4 [hunk 3]
```diff
@@ -133,9 +133,7 @@ parser MainParserImpl(
     state start {
         main_meta.sa_index = 1; // just for exmaple, used for encrypt
 
-        // TODO: can't find  better indication of recirc in the existing pna.p4
-        // This should be a field in istd or and extern
-        transition select(istd.loopedback) {
+        transition select(istd.recirculated) {
```
Comment (jfingerh) [inline examples/ipsec-acc.p4:136]: @pbhide See if this is closer to what you were hoping for, and justifies deleting the TODO comment you had.

**Comment by jfingerh** [comment]
@mariobaldi Please take a look when you get a chance.  Some of this is minor typo fixes, some is more substantive changes that enable the example P4 programs to compile without errors (e.g. see the changes to the SelectByDirection extern function definition and how it is used in the examples).

**Review by mariobaldi — APPROVED** [review]
All these changes look good.

**Pull request merged**